File: src/types/pdf-template.ts
export interface PageSettings {
  size: "A4" | "A3" | "LETTER";
  padding: {
    top: number;
    right: number;
    bottom: number;
    left: number;
  };
  fontSize: number;
}

export interface HeaderSettings {
  showLogo: boolean;
  logoUrl?: string;
  logoPosition: "left" | "center" | "right";
  logoSize: number;
  title: string;
  titleFontSize: number;
  showValidity: boolean;
  showCompanyInfo: boolean;
  companyName: string;
  companyAddress: string;
  companyPhone: string;
  companyEmail: string;
  companyWebsite: string;
  companyTaxNumber: string;
  companyInfoFontSize: number;
}



export interface ColumnSettings {
  key: string;
  label: string;
  show: boolean;
}

export interface LineTableSettings {
  columns: ColumnSettings[];
}

export interface TotalsSettings {
  showGross: boolean;
  showDiscount: boolean;
  showTax: boolean;
  showNet: boolean;
}

export interface CustomTextField {
  id: string;
  label: string;
  text: string;
  position: "header" | "footer" | "before-table" | "after-table";
  style?: {
    fontSize?: number;
    align?: "left" | "center" | "right";
    bold?: boolean;
    color?: string;
  };
}

export interface NotesSettings {
  intro?: string;
  introFontSize: number;
  footer?: string;
  footerFontSize: number;
  customFields?: CustomTextField[];
}

export interface TemplateSchema {
  page: PageSettings;
  header: HeaderSettings;
  lineTable: LineTableSettings;
  totals: TotalsSettings;
  notes: NotesSettings;
}

export interface PdfTemplate {
  id: string;
  name: string;
  type: "quote" | "invoice" | "proposal";
  locale: "tr" | "en";
  schema_json: TemplateSchema;
  version: number;
  is_default: boolean;
  created_at: string;
  updated_at: string;
  created_by?: string;
}

export interface QuoteData extends Record<string, unknown> {
  id?: string;
  number: string;
  title: string;
  description?: string;
  customer?: {
    name: string;
    company?: string;
    email?: string;
    mobile_phone?: string;
    office_phone?: string;
    address?: string;
    tax_number?: string;
    tax_office?: string;
  };
  employee?: {
    first_name: string;
    last_name: string;
    email: string;
    phone?: string;
  };
  prepared_by?: string;
  company?: {
    name: string;
    address?: string;
    phone?: string;
    email?: string;
    tax_number?: string;
    tax_office?: string;
    logo_url?: string;
    website?: string;
  };
  items: Array<{
    id: string;
    description: string;
    quantity: number;
    unit_price: number;
    unit?: string;
    tax_rate?: number;
    discount_rate?: number;
    total: number;
  }>;
  subtotal: number;
  total_discount?: number;
  total_tax: number;
  total_amount: number;
  currency: string;
  valid_until?: string;
  payment_terms?: string;
  delivery_terms?: string;
  warranty_terms?: string;
  notes?: string;
  created_at: string;
}

export interface PdfExportOptions {
  templateId?: string;
  filename?: string;
  uploadToStorage?: boolean;
  storagePath?: string;
}

File: lib/models/pdf_template.dart
class PdfTemplate {
  final String id;
  final String name;
  final String type;
  final String? locale;
  final Map<String, dynamic>? schemaJson;
  final int version;
  final bool isDefault;
  final DateTime createdAt;
  final DateTime updatedAt;
  final String? companyId;
  final String? createdBy;

  PdfTemplate({
    required this.id,
    required this.name,
    required this.type,
    this.locale,
    this.schemaJson,
    required this.version,
    required this.isDefault,
    required this.createdAt,
    required this.updatedAt,
    this.companyId,
    this.createdBy,
  });

  factory PdfTemplate.fromJson(Map<String, dynamic> json) {
    return PdfTemplate(
      id: json['id'] ?? '',
      name: json['name'] ?? '',
      type: json['type'] ?? '',
      locale: json['locale'],
      schemaJson: json['schema_json'] != null
          ? Map<String, dynamic>.from(json['schema_json'])
          : null,
      version: json['version'] ?? 1,
      isDefault: json['is_default'] ?? false,
      createdAt: DateTime.parse(json['created_at'] ?? DateTime.now().toIso8601String()),
      updatedAt: DateTime.parse(json['updated_at'] ?? DateTime.now().toIso8601String()),
      companyId: json['company_id'],
      createdBy: json['created_by'],
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'id': id,
      'name': name,
      'type': type,
      'locale': locale,
      'schema_json': schemaJson,
      'version': version,
      'is_default': isDefault,
      'created_at': createdAt.toIso8601String(),
      'updated_at': updatedAt.toIso8601String(),
      'company_id': companyId,
      'created_by': createdBy,
    };
  }
}
